scripts/west_commands/sbom/git_info_detector.py

```diff
@@ -0,0 +1,143 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+import sys
+from pathlib import Path
+from west import log
+from args import args
+from common import SbomException, command_execute, concurrent_pool_iter
+from data_structure import Data, FileInfo, Package  # pylint: disable=unused-import
+                                                    # FileInfo is used.
+
+
+def split_lines(text: str) -> 'tuple[str]':
+    '''Split input text into stripped lines removing empty lines.'''
+    return tuple(line.strip() for line in text.split('\n') if len(line.strip()))
+
+
+def get_remote_url(path: Path, remote_name: str) -> str:
+    '''Returns URL from specified git remote name and path.'''
+    output = command_execute(args.git, 'remote', 'get-url', remote_name, cwd=path)
+    return output.strip()
+
+
+def get_origin(absolute_path: Path, relative_path: Path) -> 'str|None':
+    '''Try to find out the git original source on specified path.'''
+    output, error_code = command_execute(args.git, 'remote', cwd=absolute_path,
+                                         return_error_code=True, allow_stderr=True)
+    if error_code != 0:
+        log.wrn(f'Directory "{relative_path}" does not provide valid git remote information.')
+        return None
+    remotes = split_lines(output)
+    if len(remotes) == 1:
+        return get_remote_url(absolute_path, remotes[0])
+    for remote in remotes:
+        if remote == 'origin':
+            return get_remote_url(absolute_path, remote)
```

Comment:
It seems to me that `origin` will not always be the right remote.
Consider moving this after executing the `west list` command and checking whether sha exists in the remote.

Reply:
This makes sense, done.